src/js/facility-locator/utils/helpers.js

```diff
@@ -0,0 +1,79 @@
+export const keyMap = {
+  TAB: 9,
+  ENTER: 13,
+  ESCAPE: 27,
+  SPACE: 32,
+  UP: 38,
+  DOWN: 40
+};
+
+export function getDirection(code) {
+  if (code === keyMap.UP) return -1;
+  if (code === keyMap.DOWN) return 1;
+  return false;
+}
+
+export function getSelection(options, queryType) {
+  /* eslint-disable no-param-reassign */
+  let selection = options[0];
+  if (queryType) {
+    const selectionType = queryType;
+    options = options.filter(item => !!item);
```

Comment:
Do you need to overwrite options? Is that something the callers of this function rely on?

Reply:
Options needs to be reset, so that it does not retain null references once facility is switched from benefits (which has more service options) to vet center (which has less services), the appropriate place to do this however is in `renderSelectOptions` via `this.props.options = [];`

Reply:
Now I remember why it was set up that way, props is read only. I will pass down another function prop to reset options.